src/isIterable.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Checks whether the passed value is iterable.
+ *
+ * @func isIterable
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.18.0|v2.18.0}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols|Iteration protocol}
+ * @return {boolean}
+ * @example
+ *
+ * RA.isIterable(['arrays', 'are', 'iterable']); //=> true
+ * RA.isIterable('strings are iterable, too'); //=> true
+ * RA.isIterable([]); //=> true
+ * RA.isIterable(''); //=> true
+ * RA.isIterable(new Map()); //=> true (new in ES2015)
+ * RA.isIterable(new Set()); //=> true (new in ES2015)
+ * RA.isIterable(new Int32Array()); //=> true (new in ES2017)
+ *
+ * RA.isIterable((function* () {})()); //=> true (generator objects are both iterable and iterators)
+ * RA.isIterable(function* () {}); //=> false (generators themselves are not iterable)
+ * RA.isIterable(() => {}); //=> false (regular functions are not iterable)
+ *
+ * RA.isIterable(new Promise((resolve, reject) => {})); //=> false
+ * RA.isIterable({'not': 'plain objects'}); //=> false
+ * RA.isIterable({}); //=> false
+ * RA.isIterable(new WeakMap); //=> false (new in ES2015)
+ * RA.isIterable(new WeakSet); //=> false (new in ES2015)
+ * RA.isIterable(NaN); //=> false
+ * RA.isIterable(-0); //=> false
+ * RA.isIterable(true); //=> false
+ */
+const isIterable = val => {
```

Comment:
What about the following implementation ?
```js
import { hasIn } from 'ramda';
import curry1 from 'ramda/src/internal/_curry1';
import isFunction from './isFunction';

const isIterable = curry1((val) => {
  if (typeof Symbol === 'undefined') { return false; } 

  return hasIn(Symbol.iterator, Object(val)) && isFunction(val[Symbol.iterator]);
});
```
1.) supports currying and Ramda placeholder
2.) properly checks for Symbol support with `typeof` operator
3.) doesn't check for type of the iterator return value

Reply:
I've created https://github.com/char0n/ramda-adjunct/issues/937 to define another `isAsyncIterable` predicate. It's questionable is `isIterable` should also return `true` for async iterables. I'd say YES. But this can be added later.

Reply:
sounds good to me! updating now